File: num2.cpp
#include <iostream>
using namespace std;
void main() {
	// есть полных банок
	int n = 0; 
	cout << "full bottles: ";
	cin >> n;
// проверка на дурака
while (cin.fail() || n < 0) {
		system("cls");
		cout << "Enter INTEGER > 0" << endl;
		cin.clear();
		cin.ignore();
		cin >> n;
	}
	// пустых банок на обмен
	int m = 0; 
	cout << "empty bottles for exchange: ";
	cin >> m;
// проверка на дурака
while (cin.fail() || n < 0) {
		system("cls");
		cout << "Enter INTEGER > 0" << endl;
		cin.clear();
		cin.ignore();
		cin >> n;
	}
	// пустые банки после итерации
	int mc = n; 
	// переменная для проверки возможности обменяться
	int mcount = n; 
	// счетчик итераций (выпил + обменял)
	int count = 0;
	// оставшиеся пустые банки после обмена
	int ost = 0;
	// счетчик выпитых банок
	int drinkedBottles = 0;
	while (mcount > 0) {
		cout << string(n, 'F') << endl;
		drinkedBottles += n; 
		// выпил
		count++;
		mc = n + ost;
cout << string(mc, 'E') << endl;
		ost = mc % m;
		n = mc / m;
		// обменял
		count++;
		mc = ost;
		mcount = n;
	}
	cout << drinkedBottles << " " << count-1 << endl;
}

File: num2.swift
import Foundation

func main() {
    var n = 0
    print("full bottles: ")
    if let input = readLine() {
        if let intValue = Int(input) {
            n = intValue
        }
    }
    
    var m = 0
    print("empty bottles for exchange: ")
    if let input = readLine() {
        if let intValue = Int(input) {
            m = intValue
        }
    }
    
    var mc = n
    var mcount = n
    var count = 0
    var ost = 0
    var drinkedBottles = 0
    
    while mcount > 0 {
        print(String(repeating: "F", count: n))
        drinkedBottles += n
        count += 1
        mc = n + ost
        print(String(repeating: "E", count: mc))

        ost = mc % m
        n = mc / m
        count += 1
        mc = ost
        mcount = n
    }
    
    print(drinkedBottles, count-1)
}

main()
